src/Common/DNSResolver.cpp

```diff
@@ -87,9 +87,19 @@ static DNSResolver::IPAddresses resolveIPAddressImpl(const std::string & host)
 {
     Poco::Net::IPAddress ip;
 
-    /// NOTE: Poco::Net::DNS::resolveOne(host) doesn't work for IP addresses like 127.0.0.2
-    if (Poco::Net::IPAddress::tryParse(host, ip))
-        return DNSResolver::IPAddresses(1, ip);
+    /// NOTE:
+    /// - Poco::Net::DNS::resolveOne(host) doesn't work for IP addresses like 127.0.0.2
+    /// - Poco::Net::IPAddress::tryParse() expect hex string for IPv6 (w/o brackets)
+    if (host.starts_with('['))
+    {
+        if (Poco::Net::IPAddress::tryParse(host.substr(1, host.size() - 2), ip))
```

Comment:
Add a check or assert on host size and a check for the ending character?

Reply:
I though about this, but it should not be possible to get host that starts with '[` and does not ended with `]`.
But assert is fine, will do.

Reply:
Tests had been improved too.